geode-core/src/test/java/org/apache/geode/internal/cache/ha/HARegionQueueTest.java

```diff
@@ -183,4 +203,347 @@ public void isQueueInitializedWithWaitReturnsFalseIfNotInitializedAfterWait() th
+    HARegionQueue spy = spy(haRegionQueue);
+    int maxChunkSize = 1000;
+    addEvents();
+    doReturn(eventIDS).when(spy).getGIIEvents();
+    doReturn(false).when(spy).removeDispatchedEvents(primary, internalCache, eventIDS);
+
+    spy.doSynchronizationWithPrimary(primary, internalCache);
+
+    verify(spy, never()).getChunks(eventIDS, maxChunkSize);
+    verify(spy).removeDispatchedEvents(primary, internalCache, eventIDS);
+    assertThat(spy.hasSynchronizedWithPrimary).isFalse();
+    assertThat(spy.synchronizeWithPrimaryInProgress).isFalse();
+  }
+
+  @Test
+  public void hasSynchronizedWithPrimaryRemoveChunksIfManyGIIEvents() {
+    HARegionQueue spy = spy(haRegionQueue);
+    int maxChunkSize = 1000;
+    for (int i = 0; i < 1100; i++) {
+      eventIDS.add(mock(EventID.class));
+    }
+    createChunks();
+    doReturn(eventIDS).when(spy).getGIIEvents();
+    doReturn(chunks).when(spy).getChunks(eventIDS, maxChunkSize);
+    doReturn(true).when(spy).removeDispatchedEvents(primary, internalCache, chunk1);
+    doReturn(true).when(spy).removeDispatchedEvents(primary, internalCache, chunk2);
+
+    spy.doSynchronizationWithPrimary(primary, internalCache);
+
+    verify(spy).getChunks(eventIDS, maxChunkSize);
+    verify(spy).removeDispatchedEvents(primary, internalCache, chunk1);
+    verify(spy).removeDispatchedEvents(primary, internalCache, chunk2);
+    assertThat(spy.hasSynchronizedWithPrimary).isTrue();
+    assertThat(spy.synchronizeWithPrimaryInProgress).isFalse();
+  }
+
+  private void createChunks() {
+    chunk1.add(id1);
+    chunk2.add(id2);
+    chunks.add(chunk1);
+    chunks.add(chunk2);
+  }
+
+  @Test
+  public void hasSynchronizedWithPrimaryNotSetIfRemoveChunksFails() {
+    HARegionQueue spy = spy(haRegionQueue);
+    int maxChunkSize = 1000;
+    for (int i = 0; i < 1100; i++) {
+      eventIDS.add(mock(EventID.class));
+    }
+    createChunks();
+    doReturn(eventIDS).when(spy).getGIIEvents();
+    doReturn(chunks).when(spy).getChunks(eventIDS, maxChunkSize);
+    doReturn(true).when(spy).removeDispatchedEvents(primary, internalCache, chunk1);
+    doReturn(false).when(spy).removeDispatchedEvents(primary, internalCache, chunk2);
+
+    spy.doSynchronizationWithPrimary(primary, internalCache);
+
+    verify(spy).getChunks(eventIDS, maxChunkSize);
+    verify(spy).removeDispatchedEvents(primary, internalCache, chunk1);
+    verify(spy).removeDispatchedEvents(primary, internalCache, chunk2);
+    assertThat(spy.hasSynchronizedWithPrimary).isFalse();
+    assertThat(spy.synchronizeWithPrimaryInProgress).isFalse();
+  }
+
+  @Test
+  public void getChunksReturnsEqualSizedChunks() {
+    HARegionQueue spy = spy(haRegionQueue);
+    addEvents();
+    // add more events
+    eventIDS.add(mock(EventID.class));
+    eventIDS.add(mock(EventID.class));
+
+    Collection<List<EventID>> myChunks = spy.getChunks(eventIDS, 2);
+
+    assertThat(myChunks.size()).isEqualTo(3);
```

Comment:
Could this 3 be replaced with a calculated value, and the 2 for max chunk size extracted to a variable? I think that `eventIDS.size()/maxChunkSize` should be the correct value.